src/libANGLE/renderer/d3d/d3d11/Buffer11.cpp

```diff
@@ -383,6 +386,46 @@ angle::Result Buffer11::setData(const gl::Context *context,
     return setSubData(context, target, data, size, 0);
 }
 
+angle::Result Buffer11::setDataWithUsageFlags(const gl::Context *context,
+                                              gl::BufferBinding target,
+                                              GLeglClientBufferEXT clientBuffer,
+                                              const void *data,
+                                              size_t size,
+                                              gl::BufferUsage usage,
+                                              GLbitfield flags)
+{
+    if (clientBuffer)
+    {
+        updateD3DBufferUsage(context, gl::BufferUsage::StaticDraw);
+
+        rx::BufferUsage bufferUsage;
+        switch (target)
```

Comment:
I'm wondering how far are we from calling this an up to spec solution. The version in Vulkan has to go much bigger lengths to deal with all the possible `usage` and `flags` values.
https://github.com/shapr3d/angle/blob/0b1b7fd02b3ea121e2adc55b9584bc7daf661440/src/libANGLE/renderer/vulkan/BufferVk.cpp#L326